src/Razor/src/Microsoft.CodeAnalysis.Razor/BindTagHelperDescriptorProvider.cs

```diff
@@ -360,12 +362,25 @@ private List<TagHelperDescriptor> CreateComponentBindTagHelpers(ICollection<TagH
                     }
 
                     BoundAttributeDescriptor valueAttribute = null;
+                    BoundAttributeDescriptor expressionAttribute = null;
                     var valueAttributeName = changeAttribute.Name.Substring(0, changeAttribute.Name.Length - "Changed".Length);
+                    var expressionAttributeName = valueAttributeName + "Expression";
                     for (var j = 0; j < tagHelper.BoundAttributes.Count; j++)
                     {
                         if (tagHelper.BoundAttributes[j].Name == valueAttributeName && !tagHelper.BoundAttributes[j].IsDelegateProperty())
                         {
                             valueAttribute = tagHelper.BoundAttributes[j];
+                            
+                        }
+
+                        if (tagHelper.BoundAttributes[j].Name == expressionAttributeName)
+                        {
+                            expressionAttribute = tagHelper.BoundAttributes[j];
+                        }
+
+                        if (valueAttribute != null && expressionAttribute != null)
+                        {
+                            // We found both, so we can stop looking now
```

Comment:
Do you think there's value in type-checking the expression property? I think it would be adding a test checking what error the compiler gives if you mistype the expression property.

I think it will be non-obvious to some that `int` -> `Expression<Func<int>>`

Reply:
You already did this. Derp